### PR TITLE
Adding message component

### PR DESCRIPTION
**Pull request by xwatkins**
PR for the "basic message component" - this doesn't include the "popup message"

### CODE REVIEW

File: src/components/message.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      </button>
+    )}
+  </section>
+);
+
+Message.propTypes = {
+  /**
+   * The message body
+   */
+  children: PropTypes.element.isRequired,
```
Comment (dlrice) [inline src/components/message.jsx:34]: `children` could also be `PropTypes.string`

File: stories/Message.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { Message } from '../src/components';
+import { getLipsumSentences } from '../src/mock-data/lipsum';
+
+export default {
+  title: 'Layout|Message',
+  parameters: {
+    purposeFunction: {
+      function: '',
```
Comment (dlrice) [inline stories/Message.stories.jsx:10]: Are we leaving this empty until it's more developed?

File: src/styles/components/message.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@mixin getWarningCSS($level) {
+  @include SVGDefinition(($level));
+  .alert__content {
+    border-left: getMessageBorder($level);
+  }
+}
+
+.alert {
+  -webkit-box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.2);
+  -moz-box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.2);
```
Comment (aurel-l) [inline src/styles/components/message.scss:24]: Are those prefixes manually written or were they added by a tool? I'd say we needn't worry about those anymore. [see here for compatibility](https://caniuse.com/#feat=mdn-css_properties_box-shadow)

File: stories/Message.stories.jsx (hunk 2 above)
Comment (xwatkins) [reply stories/Message.stories.jsx]: I forgot! Now added.

File: src/styles/components/message.scss (hunk 3 above)
Comment (xwatkins) [reply src/styles/components/message.scss]: Good point - now removed

File: src/components/message.jsx (hunk 1 above)
Comment (aurel-l) [reply src/components/message.jsx:34]: `Proptype.node` then?

Comment (xwatkins) [reply src/components/message.jsx]: 👍

File: src/components/message.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  failure: 'failure',
+  success: 'success',
+  info: 'info',
+};
+
+const iconSize = '1.125em';
+
+const Message = ({ children, level, onDismiss }) => (
+  <section className={`alert alert--${level}`}>
+    <section className="alert__content">
```
Comment (aurel-l) [inline src/components/message.jsx:18]: Why a section directly inside a section and not a div? Not sure about the semantic meaning here. I'm perfectly happy with the outer section though.

File: src/components/message.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  warning: 'warning',
+  failure: 'failure',
+  success: 'success',
+  info: 'info',
+};
+
+const iconSize = '1.125em';
+
+const Message = ({ children, level, onDismiss }) => (
+  <section className={`alert alert--${level}`}>
```
Comment (aurel-l) [inline src/components/message.jsx:17]:
```suggestion
  <section className={`alert alert--${level}`} role="status">
```
For ARIA reasons, [see here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_status_role)

**Comment by xwatkins** [comment]
Think all comments have now been addressed.

**Pull request merged**